discovery-provider/src/tasks/prune_plays.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import time
+from datetime import datetime
+
+from sqlalchemy import text
+from src.tasks.celery_app import celery
+
+logger = logging.getLogger(__name__)
+
+PLAYS_ARCHIVE_TABLE_NAME = "plays_archive"
+
+# ARCHIVE_PLAYS_QUERY
+# in a sliding window, delete the oldest plays before the cutoff date
+# delete no more than MAX_BATCH
+ARCHIVE_PLAYS_QUERY = """
+    INSERT INTO
+        plays_archive
+    SELECT
+        id,
+        user_id,
+        source,
+        play_item_id,
+        created_at,
+        updated_at,
+        slot,
+        signature,
+        :current_timestamp
+    FROM
+        plays
+    WHERE
+        plays.created_at <= :cutoff_timestamp
+    ORDER BY
+        id ASC
+    LIMIT
+        :max_batch
+    """
+
+# PRUNE_PLAYS_QUERY
+# prune those same plays that were archived
```

Comment:
Also considered querying the archived table's archived_at date to ensure they match up but this should be good enough.